### PR TITLE
CBG-825: Allow session create with invalid session

### PR DESCRIPTION
**Pull request by JRascagneres**
Currently a request to create a session will fail if an invalid session cookie is provided. This is the case even if valid user credentials are sent in the request body.

In order to stop this I've added a new 'priv' which specifies that when a session auth fails we can continue to authenticate as a guest.

### CODE REVIEW

File: rest/handler.go [hunk 1]
```diff
@@ -79,9 +79,10 @@ type handler struct {
 type handlerPrivs int
 
 const (
-	regularPrivs = iota // Handler requires authentication
-	publicPrivs         // Handler checks auth but doesn't require it
-	adminPrivs          // Handler ignores auth, always runs with root/admin privs
+	regularPrivs              = iota // Handler requires authentication
+	publicPrivs                      // Handler checks auth but doesn't require it
```
Comment (bbrks) [inline rest/handler.go]: Let's change the comment to be more clear about the difference between this and the new option

```suggestion
	publicPrivs                      // Handler ensures valid auth if present, but doesn't require it
```

File: rest/handler.go [hunk 2]
```diff
@@ -79,9 +79,10 @@ type handler struct {
 type handlerPrivs int
 
 const (
-	regularPrivs = iota // Handler requires authentication
-	publicPrivs         // Handler checks auth but doesn't require it
-	adminPrivs          // Handler ignores auth, always runs with root/admin privs
+	regularPrivs              = iota // Handler requires authentication
```
Comment (bbrks) [inline rest/handler.go]:
```suggestion
	regularPrivs              = iota // Handler requires valid authentication
```

**Review by bbrks — CHANGES_REQUESTED** [review]
Minor change to comments to clarify privs behaviour

File: rest/handler.go [hunk 3]
```diff
@@ -79,8 +79,8 @@ type handler struct {
 type handlerPrivs int
 
 const (
-	regularPrivs = iota // Handler requires authentication
-	publicPrivs         // Handler checks auth but doesn't require it
+	regularPrivs = iota // Handler requires valid authentication
+	publicPrivs         // Handler ensures valid auth if present, but doesn't require it
```
Comment (bbrks) [inline rest/handler.go]: this is not the case any more after the last update 😄 

```suggestion
	publicPrivs         // Handler checks auth and falls back to guest if invalid or missing
```

**Review by bbrks — CHANGES_REQUESTED** [review]

File: rest/handler.go (hunk 3 above)
Comment (JRascagneres) [reply rest/handler.go]: Ah yes.. Haha

**Review by bbrks — APPROVED** [review]
LGTM

**Pull request merged**